bal/p2p.py

```diff
@@ -88,33 +111,49 @@ def process_response_chain(self, message):
         the response chain"""
         received_chain = message.data
         if len(received_chain) == 0:
-            print('Received zero-length chain from {}'.format(message.reply_addr[1]))
+            print(
```

Comment:
let's ignore what PEP8 suggest here to split the long line. It's much easier to understand the one line here. Or we can at least split the line with `.format`